translator/internaldata/oc_to_metrics.go

```diff
@@ -200,6 +200,11 @@ func descriptorTypeToMetrics(t ocmetrics.MetricDescriptor_Type, metric pdata.Met
 		metric.SetDataType(pdata.MetricDataTypeDoubleGauge)
 		metric.DoubleGauge().InitEmpty()
 		return pdata.MetricDataTypeDoubleGauge
+	case ocmetrics.MetricDescriptor_GAUGE_DISTRIBUTION:
+		metric.InitEmpty()
+		metric.SetDataType(pdata.MetricDataTypeDoubleHistogram)
+		metric.DoubleHistogram().InitEmpty()
+		return pdata.MetricDataTypeDoubleHistogram
```

Comment:
We also return `pdata.MetricDataTypeDoubleHistogram` for `case ocmetrics.MetricDescriptor_CUMULATIVE_DISTRIBUTION:` below.
Is that correct to treat both types the same?

Reply:
Actually, the cumulative one sets 
```go
histo.SetAggregationTemporality(pdata.AggregationTemporalityCumulative)
```
Possible values are https://github.com/open-telemetry/opentelemetry-collector/blob/0c44bc24b4c69af8a07c64bd4aa35f48500cc2a8/consumer/pdata/metric.go#L24-L28

Not sure how the "Unspecified" is treated, do we need to set it to delta here?

Reply:
Reverse conversion for reference: https://github.com/open-telemetry/opentelemetry-collector/blob/c365946ef38e4defad5d9b9809064bef5623af46/translator/internaldata/metrics_to_oc.go#L249-L254

Reply:
About the temporality value, there are other GAUGE type metrics such as MetricDescriptor_GAUGE_DOUBLE which also leave the AggregationTemporality unset in descriptorTypeToMetrics(). I was also able to get correctly formatted results in an end to end test of the change that sent metrics to Stackdriver.

Returning pdata.MetricDataTypeDoubleHistogram for both DISTRIBUTION type metrics seems to make sense based on the reverse operation in metrics_to_oc. The alternative would be creating a new MetricDataType in pdata which seems like a large change that might generate push back.

Reply:
@imccarten1 this is tricky, because indeed for GAUGE_DOUBLE we do not set temporality because the Gauge in otlp proto does not have a temporality see https://github.com/open-telemetry/opentelemetry-proto/blob/master/opentelemetry/proto/metrics/v1/metrics.proto#L153

I think the current way to "not" add temporality is wrong, if you want OTLP to support gauge histograms, please create a PR that adds a temporality called "GAUGE" in the proto, or document that missing temporality means "gauge" otherwise this is an abuse of the proto.

Reply:
I've created a pull request to add the proto value AGGREGATION_TEMPORALITY_GAUGE. Please take a look. https://github.com/open-telemetry/opentelemetry-proto/pull/236